File: Programmers/Programmers_42587.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        int answer = 0;
        LinkedList<Integer> q = new LinkedList<Integer>();
        
        for(int i = 0; i < priorities.length; i++) q.add(priorities[i]);
        
        while(!q.isEmpty()){
            boolean isCheck = false;
            int first = q.getFirst();
            for(int i = 1; i < q.size(); i++){
                if(first < q.get(i)){
                    isCheck = true;
                    break;
                }
            }
            
            if(isCheck){ //첫 번째 원소보다 큰 수가 대기목록에 있는 경우
                q.removeFirst();
                q.add(first);
                
                if(location == 0) location = q.size() - 1;
                else location--;
            }
            else{ //첫 번째 원소가 가장 큰 경우
                q.removeFirst();
                answer++;
                if(location == 0) break;
                else location--;
            }
        }
        return answer;
    }
}

File: Programmers/Programmers_42587.cpp
#include <string>
#include <vector>
#include <iostream>

using namespace std;

int solution(vector<int> priorities, int location) {
    int answer = 0;
    int size = priorities.size();
    
    while(size != 0){
        bool isCheck = false;
        int first = priorities[0];
        for(int i = 1; i < priorities.size(); i++){
            int tmp = priorities[i];
            if(first < tmp) {
                isCheck = true;
                break;
            }
        }
        
        if(isCheck){ //첫 번째 원소보다 큰 수가 존재하는 경우
            priorities.erase(priorities.begin());
            priorities.push_back(first);
            if(location == 0) location = priorities.size() - 1;
            else location--;
        }
        else { //첫 번째 원소가 가장 큰 경우
            priorities.erase(priorities.begin());
            ++answer;
            if(location == 0) break;
            else location--;
        }
        size = priorities.size();
    }
    return answer;
}